python-sdk/nuscenes/scripts/export_poses.py

```diff
@@ -0,0 +1,206 @@
+REFERENCE_COORDINATES = {
+    "boston-seaport": [42.336849169438615, -71.05785369873047],
+    "singapore-onenorth": [1.2882100868743724, 103.78475189208984],
+    "singapore-hollandvillage": [1.2993652317780957, 103.78217697143555],
+    "singapore-queenstown": [1.2782562240223188, 103.76741409301758],
+}
+
+
+def get_poses(nusc: NuScenes, scene_token: str) -> List[dict]:
+    """
+    Return all ego poses for the current scene.
+    :param nusc: The NuScenes instance to load the ego poses from.
+    :param scene_token: The token of the scene.
+    :return: A list of the ego pose dicts.
+    """
+    pose_list = []
+    scene_rec = nusc.get('scene', scene_token)
+    sample_rec = nusc.get('sample', scene_rec['first_sample_token'])
+    sd_rec = nusc.get('sample_data', sample_rec['data']['LIDAR_TOP'])
+    
+    ego_pose = nusc.get('ego_pose', sd_rec['token'])
+    pose_list.append(ego_pose)
+
+    while sd_rec['next'] != '':
+        sd_rec = nusc.get('sample_data', sd_rec['next'])
+        ego_pose = nusc.get('ego_pose', sd_rec['token'])
+        pose_list.append(ego_pose)
+
+    return pose_list
+
+
+def get_coordinate(ref_lat: float, ref_lon: float, bearing: float, dist: float) -> Tuple[float, float]:
+    """
+    Using a reference coordinate, extract the coordinates of another point in space given its distance and bearing
+    to the reference coordinate. For reference, please see: https://www.movable-type.co.uk/scripts/latlong.html.
+    :param ref_lat: Latitude of the reference coordinate in degrees, ie: 42.3368.
+    :param ref_lon: Longitude of the reference coordinate in degrees, ie: 71.0578.
+    :param bearing: The angle the target point makes with the reference coordinate in radians, clockwise from north.
+    :param dist: The distance (in meters from the reference point to the target point.
+    :return: A tuple of lat and lon.
+    """
+    lat, lon = math.radians(ref_lat), math.radians(ref_lon)
+    angular_distance = dist / EARTH_RADIUS_METERS
+    
+    target_lat = math.asin(
+        math.sin(lat) * math.cos(angular_distance) + 
+        math.cos(lat) * math.sin(angular_distance) * math.cos(bearing)
+    )
+    target_lon = lon + math.atan2(
+        math.sin(bearing) * math.sin(angular_distance) * math.cos(lat),
+        math.cos(angular_distance) - math.sin(lat) * math.sin(target_lat)
+    )
+    return math.degrees(target_lat), math.degrees(target_lon)
+
+
+def derive_latlon(location: str, poses: List[dict]) -> List[Dict[str, float]]:
+    """
+    For each pose value, extract its respective lat/lon coordinate and timestamp.
+    
+    This makes the following two assumptions in order to work:
+        1. The reference coordinate for each map is in the south-western corner.
+        2. The origin of the global poses is also in the south-western corner (and identical to 1).
+
+    :param location: The name of the map the poses correspond to, ie: 'boston-seaport'.
+    :param poses: All pose dictionaries of a scene.
```

Comment:
perhaps define what is the dict here..`Dict[str, Any]`?  What are the strings here?

Reply:
It's a standard egopose dict, so I just mention that.